src/legacy/core_plugins/elasticsearch/lib/ensure_es_version.js

```diff
@@ -114,3 +114,7 @@ export function ensureEsVersion(server, kibanaVersion) {
       return true;
     });
 }
+
+function shouldIgnoreVersionMismatch(server) {
+  return server.config().get('env.dev') && server.config().get('elasticsearch.ignoreVersionMismatch');
+}
```

Comment:
As we only want that to be available in dev mode, I think it would be a good idea to actually throw a specific error if `elasticsearch.ignoreVersionMismatch` is true but `env.dev` is false (something like `elasticsearch.ignoreVersionMismatch can only be enabled in development mode`). That make it explicit that the property in only for development. Else if enabling the property in production, a user might think that the properly is simply not implemented / not taken into account.

Reply:
Good point. I've added a check that'll throw an error if `ignoreVersionMismatch` is used in production mode.